lib/models/baseVideo.dart

```diff
@@ -22,17 +22,29 @@ abstract class BaseVideo implements ShareLinks {
   BaseVideo(this.title, this.videoId, this.lengthSeconds, this.author, this.authorId, this.authorUrl, this.videoThumbnails);
 
   @override
-  String getInvidiousLink(Server server) {
-    return '${server.url}/watch?v=$videoId';
+  String getInvidiousLink(Server server, bool isAddTimestamp) {
```

Comment:
```suggestion
  String getInvidiousLink(Server server, int? timestamp) {
```

Reply:
lenthSeconds is not the current position in the video. It's the total length of the video.

Reply:
Omg my bad didn't test that well!

Reply:
Added correct logic :)